org/postgresql/test/xa/XADataSourceTest.java

```diff
@@ -341,6 +341,29 @@ public void testRestoreOfAutoCommit() throws Exception {
         xaRes.commit(xid, true);
 
         assertTrue(!conn.getAutoCommit());
+
+        // Test true case
+	conn.setAutoCommit(true);
+
+        xid = new CustomXid(15);
+        xaRes.start(xid, XAResource.TMNOFLAGS);
+        xaRes.end(xid, XAResource.TMSUCCESS);
+        xaRes.commit(xid, true);
+
+        assertTrue(conn.getAutoCommit());
```

Comment:
Please, avoid `assertTrue`/`assertFalse`: use `assertEquals` and provide human-readable message.
For instance: "XaResource should have restored connection autocommit mode after commit of the transaction to the initial state".
In other words: good test should produce a message that is good enough to copy-paste into a bug report.

With `assertTrue` all you get is a stacktrace and if such a test fails a developer has to reverse engineer the intention behind that code.